### PR TITLE
Roundtrip the undef node 

### PR DESCRIPTION
**Pull request by halvorlinder**
This PR roundtrips the undef node and provides a unit test for it.
In addition, it updates the commit hash used for building the mlir-rvsdg dialect to include the fix of the undef operation output type.

### CODE REVIEW

**Review by phate — APPROVED** [review]
Only minor comments

**Comment by sjalander** [comment]
@phate @haved This test uses both the backend and frontend.
I suggest that we place it in test/jlm/mlir instead of either the backend or frontend.
The name of the file could be MlirRoundtripTest.cpp or JlmToMlirToJlmTest.cpp.

**Comment by phate** [comment]
@sjalander Alternatively, we could split the test in two: one for frontend and one for backend?

**Comment by sjalander** [comment]
@phate In this case we could, but that adds extra work as one then need to manually construct both the RVSDG and MLIR graph for the two individual tests. That's what we have done so far, but it is tedious work that doesn't contribute to improving the test coverage.

This wouldn't work in general for the roundtrip tests we have talked about, where one create an RVSDG graph that gets converted to MLIR and then back to RVSDG-new and a simple comparison between RVSDG and RVSD-new is made.
Such tests have the added benefit that they ensure that the backend and frontend generates consistent results.

**Comment by phate** [comment]
Place them in `test/jlm/mlir` then.

File: jlm/mlir/frontend/MlirToJlmConverter.cpp [hunk 1]
```diff
@@ -308,6 +308,15 @@ MlirToJlmConverter::ConvertOperation(
     return ConvertCmpIOp(ComOp, inputs, integerType.getWidth());
   }
 
+  // Undef operation
+  else if (auto UndefOp = ::mlir::dyn_cast<::mlir::jlm::Undef>(&mlirOperation))
+  {
+    auto type = UndefOp.getResult().getType();
+    std::shared_ptr<jlm::rvsdg::Type> jlmType = ConvertType(type);
```
[the remark this answers is not in the document]
Comment (halvorlinder) [reply jlm/mlir/frontend/MlirToJlmConverter.cpp:315]: If I remove the explicit type annotation, I get 

no instance of overloaded function "jlm::llvm::UndefValueOperation::Create" matches the argument list
argument types are: (jlm::rvsdg::Region, std::unique_ptr<jlm::rvsdg::Type, std::default_delete<jlm::rvsdg::Type>>)

It seems like the issue is that getType returns a unique pointer, while ConvertType expects a shared pointer

[the remark this answers is not in the document]
Comment (phate) [reply jlm/mlir/frontend/MlirToJlmConverter.cpp:315]: Right. The ConvertType needs to be eventually fixed for this. That belongs in another PR though.

**Review by sjalander — APPROVED** [review]

**Pull request merged**